common/combat_tactics/00_combat_tactics.txt

```diff
@@ -717,89 +713,98 @@ harras_swarm_tactic = {
 
 shieldwall_tactic = {
 	days = 12
-	sprite = 2 # index of icon
+	sprite = 10
 	group = defensive
 
 	trigger = {
 		phase = skirmish
 		OR = {
-			heavy_infantry = 0.2
-			light_infantry = 0.2
-			pikemen = 0.2
+			heavy_infantry_units_020_tactic_trigger = yes
+			light_infantry_units_020_tactic_trigger = yes # replaces vanilla pikemen
+		}
+		OR = {
+			light_infantry_units_005_tactic_trigger = yes
+			archers_units_005_tactic_trigger = yes
 		}
-		archers = 0.01
 	}
 
 	mean_time_to_happen = {
 		days = 3
-		modifier = {
+		modifier = { # replaces vanilla pikemen
 			factor = 3
 			flank_has_leader = yes
 			leader = {
 				martial = 12
 			}
-			heavy_troops = {
-				who = pikemen
-				value = 0.3
-			}
+			light_infantry_troops_030_tactic_trigger = yes
 		}
 		modifier = {
 			factor = 3
 			flank_has_leader = yes
 			leader = {
 				martial = 12
 			}
-			heavy_troops = {
-				who = heavy_infantry
-				value = 0.3
-			}
+			heavy_infantry_troops_030_tactic_trigger = yes
 		}
-
-		modifier = {
+		modifier = { # replaces vanilla pikemen
 			factor = 3
 			flank_has_leader = yes
 			leader = {
 				martial = 16
 			}
-			heavy_troops = {
-				who = pikemen
-				value = 0.3
-			}
+			light_infantry_troops_030_tactic_trigger = yes
 		}
 		modifier = {
 			factor = 3
 			flank_has_leader = yes
 			leader = {
 				martial = 16
 			}
-			heavy_troops = {
-				who = heavy_infantry
-				value = 0.3
-			}
+			heavy_infantry_troops_030_tactic_trigger = yes
 		}
 	}
+	
+	# light_infantry # replaces vanilla pikemen
+	light_infantry_defensive = 2.4
+	archmage_defensive = 2.4
+
+	# heavy_infantry # average of vanilla light_infantry and heavy_infantry
+	heavy_infantry_defensive = 2.1
+	# undead_defensive = 2.1		# No shields
```

Comment:
I would argue undead are quite capable of using shields, assuming they also wield weapons; they're mindless, but not that mechanically inept. Same with thralls. Swarms is only one that makes sense to me. Mostly depends on how we're classifying Undead and Thralls in terms of actual identity.
![image](https://user-images.githubusercontent.com/32686241/65893074-5d72e380-e375-11e9-96c5-d49bd1c2cef1.png)

Reply:
BUT! Swarm got chitin! So... it makes sense to return buff to them all.